src/libraries/Microsoft.Extensions.DependencyInjection/src/ScopePool.cs

```diff
@@ -43,6 +49,9 @@ public bool Return(State state)
                 return true;
             }
 
+            // Clamp the size between the min and max
+            _maxScopeSize = Math.Min(MaxCapacity, Math.Max(size, _maxScopeSize));
```

Comment:
```suggestion
            _maxScopeSize = Math.Clamp(size, _maxScopeSize, MaxCapacity);
```

Reply:
Didn't it because it's not on .NET Standard 2.0 (and I didn't want to cross compile).